lab-jaren/app/service/player-service.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const angular = require('angular');
+
+angular.module('cfudApp')
+.service('playerService', ['$window', 'mapService', function($window, mapService) {
+  let saveState = () => {
+    try {
+      $window.localStorage.player = JSON.stringify(player);
+    } catch(e){}
+  };
+  let getState = () => {
+    try {
+      let result = JSON.parse($window.localStorage.player);
+      console.log('result', result);
+      return result;
+    } catch(e) {
+      return {};
+    }
+  };
+  let resetState = () => {
+    try {
+      $window.localStorage.clear();
+    } catch(e){}
+  };
+
+  let player = getState();
+  console.log('player', player);
+  if(!player.win)
```

Comment:
I was thinking about your refactor so many if statements, and I have an idea. It doesn't really save you lines of code, but it'll be prettier-looking.

You can just abstract your ``player`` properties into its own object literal, i.e.:

```
let defaultPlayer = {
  player.win: false,
  player.name: 'You',
  player.location: 'mapService[location]',
  player.history: [],
etc...
}
```

And then starting above on Line 27, your code will be:

```
let player = getState();
if (player === {}) player = defaultPlayer
```

Again, not really a huge saver for lines, but it'll be much better looking than 20-ish if statements that you'd otherwise have. I hope this helps!